.github/workflows/release.yml

```diff
@@ -14,21 +14,24 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+
     - name: Set up JDK 17
       uses: actions/setup-java@v3
       with:
         java-version: '17'
         distribution: 'temurin'
+
     - name: Setup Android SDK
       uses: android-actions/setup-android@v3
-    - name: Build with Gradle
-      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
-      with:
-        arguments: build check
+
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3
+    
+    - name: Build and Check
+      run: ./gradlew build check
+
     - name: Publish to Plugin Portal
-      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
-      with:
-        arguments: publishPlugins -Pgradle.publish.key=$GRADLE_PUBLISH_KEY -Pgradle.publish.secret=$GRADLE_PUBLISH_SECRET
+      run: ./gradlew publishPlugins -Pgradle.publish.key=$GRADLE_PUBLISH_KEY -Pgradle.publish.secret=$GRADLE_PUBLISH_SECRET
```

Comment:
It seems weird to me to see an invocation of Gradle making release artifacts without `--no-daemon` - do you know if Gradle daemons would be cleared in-between builds?

Reply:
Oh I'm not familiar enough with what issues might arise. We can use --no-daemon if that makes sense here.

Reply:
alright, this should just all happen together.

Reply:
talked offline and you mentioned that the machines were isolated so there shouldn't be leftover memory state from previous runs